src/System.IO.Compression.ZipFile/src/System/IO/Compression/ZipFile.cs

```diff
@@ -569,8 +570,11 @@ private static string EntryFromPath(string entry, int offset, int length)
 
         private static Boolean IsDirEmpty(DirectoryInfo possiblyEmptyDir)
         {
-            foreach (FileSystemInfo fi in possiblyEmptyDir.EnumerateFileSystemInfos("*", SearchOption.AllDirectories))
-                return false;
+            using (IEnumerator<String> enumerator = Directory.EnumerateFileSystemEntries(possiblyEmptyDir.FullName).GetEnumerator())
+            {
+                if (enumerator.MoveNext())
+                    return false;
+            }
 
             return true;
```

Comment:
I had it like the way you suggest before I submitted the PR, but I checked http://sourceof.net quickly to see the style used for [`Enumerable.Any`](https://github.com/Microsoft/referencesource/blob/74706335e3b8c806f44fa0683dc1e18d3ed747c2/System.Core/System/Linq/Enumerable.cs#L1168), so I changed it to match that. However, I just checked the latest corefx source, and `Enumerable.Any` was [simplified](https://github.com/dotnet/corefx/blob/6b5a8aed6b8cc87f074024d7947236ca2a769c2d/src/System.Linq/src/System/Linq/Enumerable.cs#L1463) as you suggest, so I'll go ahead and simplify it here as well.